### ISSUE TITLE
Firebase-less flavour / self-hostable flavour

### ISSUE THREAD

**Issue by alexanderadam** [veertly/app#68]
I know that this isn't currently on your roadmap but it would be great if there would be a Firebase-less flavour that would allow a self hosted installation. Maybe even a Docker image.

I don't expect you to solve this anytime soon but maybe somebody else want to give it a go.
This way this issue can track anything about it (or maybe you will even change your mind about firebase). :wink:

**Comment by joaoaguiam** [comment]
Hi @alexanderadam , thanks for creating this issue. 

To be honest the effort is quite high in order to have a real-time application as we have now. But if anyone wants to work on it I can help them understand the architecture and on a possible solution for it.

**Comment by aspiers** [comment]
I agree that this would be a great thing for the community (which would also benefit the company by bringing more developers in), and it's a good idea to use this issue to consolidate and track any efforts in that direction.

**Comment by veerjainATgmail** [comment]
> Hi @alexanderadam , thanks for creating this issue.
> 
> To be honest the effort is quite high in order to have a real-time application as we have now. But if anyone wants to work on it I can help them understand the architecture and on a possible solution for it.

Would love to attempt this. Any pointers on how to proceed? thanks.

**Comment by rasos** [comment]
I think veertly would really be adding value to existing open source based video conference solutions. Two documents would be helpful to motivate people playing around with veertly and let them create pull requests on additional features or improvements: 

1. how to install veertly (even using firebase) on your own server
2. short architecture overview

**Comment by joaoaguiam** [comment]
Hi @rasos 

For point 1, you can check the pipeline that builds and deploys Veertly on Firebase. You just need to setup your firebase before. The services used are Authentication, Firestore, Realtime Database, Functions and Storage.

We don't have any architecture document but it is straightforward:
- front-end in react that communicates directly with Firestore database
- some functions are deployed and called from the front-end, or when some documents are updated on the firestore
- storage to store users avatars and events banners
- authentication to keep track of the user sessions.

I am happy to jump on a call if you want, ping me on telegram.

**Comment by rasos** [comment]
We have been able to deploy the veertly app on firebase for testing.

Which firestore rules are recommended? We use: 

```
rules_version = '2';
service cloud.firestore {
  match /databases/{database}/documents {
    match /{document=**} {
      allow read, write: if true;
    }
  }
}
```

**Comment by alexanderadam** [comment]
Maybe this issue will be solved easier with @supabase ( https://supabase.io/ ).

**Comment by joaoaguiam** [comment]
Indeed that could be a possibility, but didn't investigate enough to say how much complicated it will be.

**Comment by joaoaguiam** [comment]
> We have been able to deploy the veertly app on firebase for testing.
> 
> 
> 
> Which firestore rules are recommended? We use: 
> 
> 
> 
> ```
> 
> rules_version = '2';
> 
> service cloud.firestore {
> 
>   match /databases/{database}/documents {
> 
>     match /{document=**} {
> 
>       allow read, write: if true;
> 
>     }
> 
>   }
> 
> }
> 
> ```

If you need any support on the rules just ping me on telegram, happy to help.

**Comment by k1n6b0b** [comment]
I currently use the docker deployment of Jitsi and would love to use this adaptation. I checked out the demo and was impressed with the layout - as we all look to new ways to manage social interaction the concept of breakout rooms, or sidebar conversations are a key fix to the single-conversation style of a 'meeting room'.

I wont be much help besides hoo ra ra'ing this and saying thanks!

I'd want to self-host (and could try firebase but I generally am ignorant and avoidant of google proprietary technology stacks)

**Comment by joaoaguiam** [comment]
Thanks for the kind words @k1n6b0b ! 

Veertly doesn't work on a completely self-hosted environment, it is based on Firebase so you will need to create a project and configure the code to connect to your project. Ping me on telegram if you need any support.

**Comment by rpghawke** [comment]
Greetings @joaoaguiam we have a non-profit 100% volunteer-run charity using games for special needs populations. I was hoping to see if there was a way we could self-host Veertly. We can't be in the cloud because some of the populations are protected under HIPAA and the like so we must host on our own local servers for privacy.  I'm not sure where else to contact you for some suggestions on options for getting this started.  We would gladly share back our efforts the opensource community, everything we do is very open source.  Please advise. Thank you!
